packages/mobile/src/home/CashInBottomSheet.test.tsx

```diff
@@ -1,12 +1,90 @@
-import { fireEvent, render } from '@testing-library/react-native'
+import { fireEvent, render, waitFor } from '@testing-library/react-native'
+import { FetchMock } from 'jest-fetch-mock/types'
 import * as React from 'react'
 import { Provider } from 'react-redux'
+import { PaymentMethod } from 'src/fiatExchanges/FiatExchangeOptions'
 import CashInBottomSheet from 'src/home/CashInBottomSheet'
 import { navigate } from 'src/navigator/NavigationService'
+import { navigateToURI } from 'src/utils/linking'
 import { Screens } from 'src/navigator/Screens'
 import { createMockStore } from 'test/utils'
 
+const mockRampProvider = {
+  name: 'Ramp',
+  restricted: false,
+  paymentMethods: [PaymentMethod.Card, PaymentMethod.Bank],
+  url: 'www.fakewebsite.com',
+  logo:
+    'https://firebasestorage.googleapis.com/v0/b/celo-mobile-mainnet.appspot.com/o/images%2Framp.png?alt=media',
+  quote: [
+    { paymentMethod: PaymentMethod.Card, digitalAsset: 'cusd', returnedAmount: 100, fiatFee: 0 },
+  ],
+  cashIn: true,
+  cashOut: false,
+  unavailable: false,
+}
+
+const mockRampProviderUnavailable = {
+  name: 'Ramp',
+  restricted: false,
+  paymentMethods: [PaymentMethod.Card, PaymentMethod.Bank],
+  url: 'www.fakewebsite.com',
+  logo:
+    'https://firebasestorage.googleapis.com/v0/b/celo-mobile-mainnet.appspot.com/o/images%2Framp.png?alt=media',
+  quote: [
+    { paymentMethod: PaymentMethod.Card, digitalAsset: 'cusd', returnedAmount: 100, fiatFee: 0 },
+  ],
+  cashIn: true,
+  cashOut: false,
+  unavailable: true,
+}
+
+const mockRampProviderRestricted = {
+  name: 'Ramp',
+  restricted: true,
+  paymentMethods: [PaymentMethod.Card, PaymentMethod.Bank],
+  url: 'www.fakewebsite.com',
+  logo:
+    'https://firebasestorage.googleapis.com/v0/b/celo-mobile-mainnet.appspot.com/o/images%2Framp.png?alt=media',
+  quote: [
+    { paymentMethod: PaymentMethod.Card, digitalAsset: 'cusd', returnedAmount: 100, fiatFee: 0 },
+  ],
+  cashIn: true,
+  cashOut: false,
+  unavailable: false,
+}
+
+const mockRampProviderNoCashIn = {
+  name: 'Ramp',
+  restricted: false,
+  paymentMethods: [PaymentMethod.Card, PaymentMethod.Bank],
+  url: 'www.fakewebsite.com',
+  logo:
+    'https://firebasestorage.googleapis.com/v0/b/celo-mobile-mainnet.appspot.com/o/images%2Framp.png?alt=media',
+  quote: [
+    { paymentMethod: PaymentMethod.Card, digitalAsset: 'cusd', returnedAmount: 100, fiatFee: 0 },
+  ],
+  cashIn: false,
+  cashOut: false,
+  unavailable: false,
+}
```

Comment:
```suggestion
const mockRampProviderNoCashIn = {
  ...mockRampProvider,
  cashIn: false
}
```
optional nit. could help readability